docx/oxml/text/run.py

```diff
@@ -137,34 +137,43 @@ def text(self):
         equivalent.
         """
         text = ''
-        if CT_FldChar.numOfNestedFldChar > 0:
-            for child in self:
-                if child.tag == qn('w:fldChar'):
-                    if child.fldCharType == 'begin':
-                        CT_FldChar.numOfNestedFldChar += 1
-                    else:
-                        # `w:fldChar` stores instruction text from `w:fldCharType='begin'`
-                        # to `w:fldCharType='separate'` if 'separate' exists, if not then
-                        # until `w:fldCharType='end'`.
-                        CT_FldChar.numOfNestedFldChar -= 1
-        else:
-            for child in self:
-                if child.tag == qn('w:t'):
-                    t_text = child.text
-                    text += t_text if t_text is not None else ''
-                elif child.tag == qn('w:tab'):
-                    text += '\t'
-                elif child.tag == qn('w:br'):
-                    text += '\n'
-                elif child.tag == qn('w:cr'):
-                    text += '\r'
-                elif child.tag == qn('w:noBreakHyphen'):
-                    text += '-'
-                elif child.tag == qn('w:sym'):
-                    text += child.readSymbol
-                elif child.tag == qn('w:fldChar'):
-                    if child.fldCharType == 'begin':
-                        CT_FldChar.numOfNestedFldChar += 1
+        child_text = ''
```

Comment:
I think you can move this line to be the first line in the for loop and then you don't need the line 158 (you don't need to reset `child_text` after setting it)? I'm fine with this version as well, whichever you prefer.